content/blog/2016/2016-11-20-what-jvm-versions-are-running-jenkins-the-return.adoc

```diff
@@ -0,0 +1,46 @@
+
+Like for link:https://jenkins.io/blog/2015/11/03/what-jvm-versions-are-running-jenkins/[last year's article about the same subject], yet another link:https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/jenkinsci-dev/fo5nKLhZK5U/Zb9jHpBJBQAJ[recent discussion about finally requiring Java 8 to run future versions Jenkins] pushed me to gather some more factual data around it.
+
+TIP: All numbers shown below are derived from the new `link:stats.jenkins.io/plugin-installation-trend/jvms.json[jvms.json]` file now generated automatically every months, after the two related pull-requests  link:https://github.com/jenkins-infra/infra-statistics/pull/21[1] and link:https://github.com/jenkins-infra/infra-statistics/pull/22[2] got merged.
+footnote:[You are more than welcome to review those Pull-Requests and shout if you see something wrong in the calculations.]
+
+== Java 8 now the most used version, and growing
+
+If we look at the global numbers, Java 8 runtimes now represent 52,6% of the Jenkins instances running in the world
+footnote:[And not having opted out of the statistics push].
+
+image::/images/2016-jvm-stats-all.png[]
+
+And if you look at the trend, Java 8 is clearly growing fast.
+
+=== Zooming into the Jenkins 2.x instances subset
+
+Now, if you look at that picture, though already interesting and showing a clear trend towards Java 8 runtime adoption, some might argue it's being too nice to older JREs.
+The reasoning could be: instances running (very) old Jenkins versions may not be the ones you want to look at when trying to plan the future of an opensource project:
+those are indeed probably not going to upgrade in general anyway, or when they do, upgrading the JRE would be a small thing compared to the rest to be tested with such a gap.
+
+So, if we only keep the instances running Jenkins 2.x, then the proportion of Java 8 goes to almost 70% compared to Java 7 (Jenkins 2.x requires Java 7)
+footnote:[69% for October, 67% in September]:
+
+image::/images/2016-jvm-stats-only-2.x.png[]
+
+== Conclusion
+
+Java 8 adoption numbers are getting bigger, while every other JREs are going down.
+
+If you are still using a JRE 7 to run Jenkins, it is seriously time to think about upgrading to 8.
+Knowing that it's definitely not a bleeding-edge path might help you go that way, especially if you generally do not like upgrades.
+Also, as a reminder, the most used JDK, Oracle JDK 7 now got end-of-lifed more than 18 months ago.
+
+Contrary to the past attempts the previous years, the discussion on the Jenkins development mailing list did not trigger strong rebutals by many people.
+So, maybe, time has come.
```

Comment:
This is kind of an ominous last sentence, perhaps something more cheery

> Perhaps it's finally time for Mr. Jenkins to upgrade to Java 8!


lolidunno